aten/src/ATen/cpu/vec/vec256/vec256_mask.h

```diff
@@ -46,14 +101,97 @@ struct VecMaskCast<int, 1, float, 1> {
     return Vectorized<int>(_mm256_castps_si256(vec_mask[0]));
   }
 };
+template <>
+struct VecMaskCast<float, 2, int, 2> {
+  static inline VecMask<float, 2> apply(const VecMask<int, 2>& vec_mask) {
+    return VectorizedN<float, 2>(_mm256_castsi256_ps(vec_mask[0]), _mm256_castsi256_ps(vec_mask[1]));
+  }
+};
+template <>
+struct VecMaskCast<int, 2, float, 2> {
+  static inline VecMask<int, 2> apply(const VecMask<float, 2>& vec_mask) {
+    return VectorizedN<int, 2>(_mm256_castps_si256(vec_mask[0]), _mm256_castps_si256(vec_mask[1]));
+  }
+};
```

Comment:
Instead of specializing sizes with `2`, can we generalize it with arbitrary `N` with a for loop calling the specialized size `1`?

Reply:
Revised according to suggestions